### PR TITLE
Use "Smalltalk vm wordSize" to determine the version of the VM: 32 or 64

### PR DESCRIPTION
**Pull request by VincentBlondeau**
Change isWin32 sends to isWindows to includes the isWin64.
Tested on Win64

Fixes:
#6709
#2246
#5968

### CODE REVIEW

**Comment by MarcusDenker** [comment]
there are some tests failing

windows-64 / Tests-windows-64 / testKeyBindings – Windows64.Spec2.Backend.Tests.SpTextAdapterTest
windows-64 / Tests-windows-64 / testKeyBindings – Windows64.Spec2.Code.Backend.Tests.SpCodeAdapterTest
windows-64 / Tests-windows-64 / testWhenSubmitDo – Windows64.Spec2.Tests.Core.Widgets.SpCodePresenterTest
windows-64 / Tests-windows-64 / testWhenSubmitDo – Windows64.Spec2.Tests.Core.Widgets.SpLabelledTextInputTest
windows-64 / Tests-windows-64 / testWhenSubmitDo – Windows64.Spec2.Tests.Core.Widgets.SpNumberInputFieldPresenterTest
windows-64 / Tests-windows-64 / testWhenSubmitDo – Windows64.Spec2.Tests.Core.Widgets.SpTextInputFieldPresenterTest
windows-64 / Tests-windows-64 / testWhenSubmitDo – Windows64.Spec2.Tests.Core.Widgets.SpTextPresenterTest

File: src/System-OSEnvironments/Win32Environment.class.st [hunk 1]
```diff
@@ -19,8 +19,9 @@ Class {
 }
 
 { #category : #testing }
-Win32Environment class >> isDefaultFor: aPlatform [ 
-	^ aPlatform isWin32
+Win32Environment class >> isDefaultFor: aPlatform [
```
Comment (dionisiydk) [inline src/System-OSEnvironments/Win32Environment.class.st:22]: What about rename it to WinEnvironment?

File: src/System-OSEnvironments/Win32Environment.class.st [hunk 2]
```diff
@@ -19,8 +19,9 @@ Class {
...
+
+	^ aPlatform isWin32 or: [ aPlatform isWin64 ]
```
Comment (dionisiydk) [inline src/System-OSEnvironments/Win32Environment.class.st:24]:
```suggestion
	^ aPlatform isWindows
```

**Review by dionisiydk — APPROVED** [review]

**Comment by dionisiydk** [comment]
Tests are unrelated

**Comment by MarcusDenker** [comment]
Failing tests seem related to the change: exactly  windows related tests start to fail with this change

**Comment by astares** [comment]
@VincentBlondeau would you be able to comment on/adopt to the suggested changes from @dionisiydk so this issue could be moved forward?

**Comment by dionisiydk** [comment]
Now there is a merge problem. 
But anyway I did not intent to block the PR: just a possible improvements which can be done later.

File: src/Alien-Core/CallbackForWin64X64.class.st [hunk 3]
```diff
@@ -14,11 +14,8 @@ CallbackForWin64X64 class >> abi [
 
 { #category : #'class initialization' }
 CallbackForWin64X64 class >> isForCurrentPlatform [
-	| processor |
-	(#('Win32' 'Win64') includes: (SmalltalkImage current getSystemAttribute: 1001)) ifFalse:
-		[^false].
-	processor := (SmalltalkImage current getSystemAttribute: 1003) asLowercase.
-	^processor first = $x and: [processor = 'x64' or: [('x86#64*' match: processor)]]
+
+	^ Win64Platform isActivePlatform
```
Comment (guillep) [inline src/Alien-Core/CallbackForWin64X64.class.st:18]: This class does not exist anymore in Pharo10

File: src/Multilingual-Languages/Latin1Environment.class.st [hunk 4]
```diff
@@ -38,7 +38,7 @@ Latin1Environment class >> supportedLanguages [
 
 { #category : #'subclass responsibilities' }
 Latin1Environment class >> systemConverterClass [
-	Smalltalk os isWin32
+	Smalltalk os isWindows
 		ifTrue: [^ UTF8TextConverter].
  	Smalltalk os isMacOSX
 		ifTrue: [ ^ UTF8TextConverter ].
```
Comment (guillep) [inline src/Multilingual-Languages/Latin1Environment.class.st:44]: This class is deprecated in Pharo10

File: src/Multilingual-Languages/Latin2Environment.class.st [hunk 5]
```diff
@@ -38,7 +38,7 @@ Latin2Environment class >> supportedLanguages [
 { #category : #'subclass responsibilities' }
 Latin2Environment class >> systemConverterClass [
 
-	Smalltalk os isWin32 
+	Smalltalk os isWindows
```
Comment (guillep) [inline src/Multilingual-Languages/Latin2Environment.class.st:41]: This class is deprecated in Pharo10

File: src/Multilingual-Languages/Latin9Environment.class.st [hunk 6]
```diff
@@ -30,7 +30,7 @@ Latin9Environment class >> supportedLanguages [
 { #category : #'subclass responsibilities' }
 Latin9Environment class >> systemConverterClass [
 
-	Smalltalk os isWin32 
+	Smalltalk os isWindows
```
Comment (guillep) [inline src/Multilingual-Languages/Latin9Environment.class.st:33]: This class is deprecated in Pharo10

File: src/Multilingual-OtherLanguages/GreekEnvironment.class.st [hunk 7]
```diff
@@ -31,7 +31,7 @@ GreekEnvironment class >> supportedLanguages [
 { #category : #'subclass responsibilities' }
 GreekEnvironment class >> systemConverterClass [
 
-	Smalltalk os isWin32
+	Smalltalk os isWindows
```
Comment (guillep) [inline src/Multilingual-OtherLanguages/GreekEnvironment.class.st:34]: This class is deprecated in Pharo10

File: src/Multilingual-OtherLanguages/JapaneseEnvironment.class.st [hunk 8]
```diff
@@ -63,7 +63,7 @@ JapaneseEnvironment class >> supportedLanguages [
 JapaneseEnvironment class >> systemConverterClass [
 	| encoding |
 
-	Smalltalk os isWin32 
+	Smalltalk os isWindows
```
Comment (guillep) [inline src/Multilingual-OtherLanguages/JapaneseEnvironment.class.st:66]: This class is deprecated in Pharo10

File: src/Multilingual-OtherLanguages/KoreanEnvironment.class.st [hunk 9]
```diff
@@ -29,7 +29,7 @@ KoreanEnvironment class >> supportedLanguages [
 { #category : #'subclass responsibilities' }
 KoreanEnvironment class >> systemConverterClass [
 	| encoding |
-	Smalltalk os isWin32 
+	Smalltalk os isWindows
```
Comment (guillep) [inline src/Multilingual-OtherLanguages/KoreanEnvironment.class.st:32]: This class is deprecated in Pharo10

**Review by guillep — APPROVED** [review]
Superseeded by https://github.com/pharo-project/pharo/pull/10008

**Comment by JanBliznicenko** [comment]
This change probably caused issue #10026

**Pull request closed without merging**